Tribler/community/tunnel/hidden_community.py

```diff
@@ -312,26 +397,34 @@ def on_key_request(self, messages):
 
     def check_key_response(self, messages):
         for message in messages:
-            if not message.source.startswith(u"circuit_"):
```

Comment:
The only problem I can come up with is that Charlie is not using the dispersy socket to send the packet back to Dean. That could cause a problem with some Nats. 
I'm not sure how to fix that.